contracts/src/interfaces/IFunctionGateway.sol

```diff
@@ -1,91 +1,56 @@
-    event ScalarUpdated(uint256 scalar);
 }
 
 interface IFunctionGatewayErrors {
-    error RequestNotFound(bytes32 requestId);
-    error ContextMismatch(bytes32 contextHash, bytes context);
-    error OutputMismatch(bytes32 outputHash, bytes context);
-    error InputsRootMismatch(bytes32 inputsRoot, bytes32[] inputHashes);
-    error OutputsRootMismatch(bytes32 outputsRoot, bytes32[] outputHashes);
-    error VerificationKeysRootMismatch(bytes32 outputsRoot, bytes32[] outputHashes);
-    error ProofNotFulfilled(bytes32 requestId);
-    error ProofAlreadyFulfilled(bytes32 requestId);
-    error InvalidProof(address verifier, bytes32 inputHash, bytes32 outputHash, bytes proof);
-    error CallbackFailed(address callbackAddress, bytes4 callbackSelector);
-    error CallbackAlreadyFulfilled(bytes32 requestId);
-    error LengthMismatch(uint256 expected, uint256 actual);
-    error InsufficientFeeAmount(uint256 expected, uint256 actual);
-    error RefundFailed(address refundAccount, uint256 refundAmount);
+    error InvalidRequest(
+        uint32 nonce,
+        bytes32 expectedRequestHash,
+        bytes32 requestHash
+    );
+    error CallbackFailed(bytes4 callbackSelector, bytes output, bytes context);
+    error InvalidCall(bytes32 functionId, bytes32 inputHash);
+    error CallFailed(address callbackAddress, bytes callbackData);
+    error InvalidProof(
+        address verifier,
+        bytes32 inputHash,
+        bytes32 outputHash,
+        bytes proof
+    );
 }
 
 interface IFunctionGateway is IFunctionGatewayEvents, IFunctionGatewayErrors {
-    function requests(bytes32 requestId)
-        external
-        view
-        returns (bytes32, bytes32, bytes32, bytes32, address, bytes4, bool, bool);
-
-    function request(bytes32 functionId, bytes memory input, bytes4 callbackSelector, bytes memory context)
-        external
-        payable
-        returns (bytes32);
-
-    function request(
-        bytes32 functionId,
-        bytes memory input,
-        bytes4 callbackSelector,
-        bytes memory context,
-        uint256 gasLimit,
-        address refundAccount
+    function zkRequest(
+        bytes32 _functionId,
+        bytes memory _input,
+        bytes memory _context,
+        bytes4 _callbackSelector,
+        uint32 _callbackGasLimit
     ) external payable returns (bytes32);
 
-    function fulfill(bytes32 requestId, bytes32 outputHash, bytes memory proof) external;
-
-    function fulfillBatch(
-        bytes32[] memory requestIds,
-        bytes memory aggregateProof,
-        bytes32 inputsRoot,
-        bytes32[] memory outputHashes,
-        bytes32 outputsRoot,
-        bytes32 verificationKeyRoot
-    ) external;
-
-    function callback(bytes32 requestId, bytes memory output, bytes memory context) external;
-
-    function calculateFeeAmount() external view returns (uint256);
-
-    function calculateFeeAmount(uint256 gasLimit) external view returns (uint256);
+    function zkCall(
```

Comment:
I think this should return `(bool ok, bytes memory)` to match the structure of `.call` normally, what do you think?